### PR TITLE
[Feature/add lift demo] Add hotel scenario

### PR DESCRIPTION
**Pull request by MakinoharaShouko**
This is the demo scenario I have been working on as part of Google Summer of Code. The map is a reduced version and the full version can be accessed [here](https://github.com/MakinoharaShouko/hotel). The full map consumes a huge amount of memory thus cannot run. This version is reduced so that the rmf_core can handle most requests. Though It still breaks when the request is long (especially for the MiR100 fleet)

### CODE REVIEW

**Review by Kevinskwk — CHANGES_REQUESTED** [review]
Thanks for the work! It looks great in the traffic_editor, but two things are stopping me from seeing this demo in gazebo:
1. As commented, the robot are now renamed. So all appearances of the branded robot names should be changed.
2. It seems that the names of the models you included are not up to date. You can try the following to update your thumbnails and models:
  - clone the latest of https://github.com/osrf/traffic_editor_assets into your `$HOME/.traffic_editor/assets/` directory
  - update the model names in your `building.yaml` file according to the latest thumbnails available
  - then do the following to download the new models:
  ```
cd ~/.gazebo/
mv models models-bak # just as a backup
mkdir models
cp -r models-bak/sun models/.
cp -r models-bak/ground_plane models/.
cd YOUR_WORKSPACE
rm -rf build/rmf_demo_maps install/rmf_demo_maps
source /opt/ros/eloquent/setup.bash
colcon guild --packages-up-to demos --cmake-args -DCMAKE_BUILD_TYPE=RELEASE
  ```

In addition, could you also add some descriptions and instructions on running the hotel demo in `README.md`? Thank you!

**Comment by Yadunund** [comment]
It would be good to also update the `README.md` file with instructions to launch this world.

File: demos/launch/hotel.launch.xml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        <env name="GAZEBO_MODEL_PATH" value="$(var model_path)" />
+        <env name="GAZEBO_RESOURCE_PATH" value="$(var resource_path)" />
+        <env name="GAZEBO_PLUGIN_PATH" value="$(var plugin_path)" />
+      </executable>
+    </group>
+  </group>
+
+  <!-- Magni fleet adapter and robot state aggregator needed for the Magni slotcar_plugin -->
+  <group>
+    <let name="fleet_name" value="magni"/>
```
[the remark this answers is not in the document]
Comment (MakinoharaShouko) [reply demos/launch/hotel.launch.xml]: I have made the name changes to the hotel map and the launch file and also added the instruction part in `README.md`. There are some model thumbnails that I generated with `thumbnail_generator` so I left them untouched.

[the remark this answers is not in the document]
Comment (Kevinskwk) [reply demos/launch/hotel.launch.xml]: By generating model thumbnails, do you mean that you have included some custom models?

[the remark this answers is not in the document]
Comment (MakinoharaShouko) [reply demos/launch/hotel.launch.xml]: Yeah

**Comment by Kevinskwk** [comment]
Thank you for the changes, I am able to launch the world now. Though there are still several minor issues, I guess I can quickly fix them after merging. Just to clarify, are all the doors and some lifts missing to reduce memory usage? Also, it would be great if you could add the link to the full version map in `README.md` so that people can still try if interested.

File: demos/launch/hotel.launch.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kevinskwk) [reply demos/launch/hotel.launch.xml]: Okay, so where are the models placed in?

[the remark this answers is not in the document]
Comment (MakinoharaShouko) [reply demos/launch/hotel.launch.xml]: I downloaded them from ignition fuel. Some are from `~/.gazebo/models`

**Comment by MakinoharaShouko** [comment]
Yeah I removed the doors and most lifts to reduce the memory usage. I have also attached the link to the full hotel repo

File: demos/launch/hotel.launch.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kevinskwk) [reply demos/launch/hotel.launch.xml]: Oh ok, then it should be fine

**Review by Kevinskwk — APPROVED** [review]
Thanks for the work. I'll merge this PR first and then quickly fix some minor issues

**Pull request merged**